common/persistence/nosql/nosqlplugin/cassandra/gocql/session_test.go

```diff
@@ -68,4 +70,14 @@ func TestSessionEmitsMetricOnRefreshThrottle(t *testing.T) {
 	metricsHandler.EXPECT().Counter(metrics.CassandraSessionRefreshFailures.Name()).Return(metrics.NoopCounterMetricFunc)
 
 	s.refresh()
+	controller.Finish()
+}
+
+func TestPanicCapture(t *testing.T) {
+	_, err := initSession(log.NewNoopLogger(), func() (*gocql.ClusterConfig, error) {
+		return &gocql.ClusterConfig{Hosts: []string{"0.0.0.0"}}, nil
+	}, metrics.NoopMetricsHandler)
+
+	assert.Error(t, err)
+	assert.Contains(t, err.Error(), "panic: no valid connect address for host")
```

Comment:
nit: We shouldn't be asserting on actual error strings provided by downstream systems. It leads to fragile tests which break when underlying library (gocql here) changes or updates the error message.
ex: if they change [this error](https://github.com/gocql/gocql/blob/master/host_source.go#L206) suddenly our tests will fail. A suggestion would be to just assert that a panic was caught.

Reply:
Well, if gocql properly handles that case and return error, this test would fail too. But I changed the test to assert it contains panic only.